core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -215,30 +213,61 @@ private static void ValidateMode(string? mode, CommandResult commandResult)
     /// Validates if the provided transport is valid.
     /// </summary>
     /// <param name="transport">The transport to validate.</param>
+    /// <param name="enableInsecureTransports">Whether insecure transports are enabled.</param>
     /// <param name="commandResult">Command result to update on failure.</param>
-    private static void ValidateTransport(string? transport, CommandResult commandResult)
+    private static void ValidateTransport(TransportTypes transport, bool enableInsecureTransports, CommandResult commandResult)
     {
-        if (transport is null || transport == TransportTypes.StdIo)
+        if (transport == TransportTypes.StdIo)
         {
-            return; // Success
+            if (enableInsecureTransports)
+            {
+                commandResult.AddError($"The --enable-insecure-transports option cannot be used with the default {TransportTypes.StdIo} transport. To use this option, specify {TransportTypes.Http} transport with --transport http.");
+            }
+            return;
+        }
+
+        if (transport == TransportTypes.Http)
+        {
+#if ENABLE_REMOTE
```

Comment:
While working on this PR, I felt the directive name could be improved - something like ENABLE_INCOMING_HTTP_AUTH or ENABLE_AUTHENTICATED_HTTP_SERVICE, andf any references to runAsRemoteService variables (and similar) could be renamed to runAsAuthenticatedHttpService for better clarity.